tests/expressions/tests.py

```diff
@@ -476,14 +476,20 @@ def test_object_update_unsaved_objects(self):
         with self.assertRaisesMessage(ValueError, msg):
             acme.save()
 
-    def test_ticket_11722_iexact_lookup(self):
+    def test_lookups_with_f_has_valid_sql(self):
+        """
+        Query lookups that reference an F() expression produce valid sql (#11722).
+        """
         Employee.objects.create(firstname="John", lastname="Doe")
         test = Employee.objects.create(firstname="Test", lastname="test")
 
         queryset = Employee.objects.filter(firstname__iexact=F("lastname"))
         self.assertSequenceEqual(queryset, [test])
 
-    def test_ticket_16731_startswith_lookup(self):
+    def test_startswith_endswith_works_with_f_expression(self):
+        """
+        startswith endswith and contains works with F expression (#16731).
+        """
```

Comment:
```suggestion
    def test_filter_startswith_with_f_expression(self):
```
(this test doesn't test endswith)